src/gui/libcloudproviders/libcloudproviders.cpp

```diff
@@ -126,7 +126,15 @@ void LibCloudProvidersPrivate::exportFolder(Folder *folder)
         g_error_free(error);
     }
 
-    auto exporter = cloud_providers_account_exporter_new(_exporter, folder->id().constData());
+    // DBus object paths must not contain characters other than [A-Z][a-z][0-9]_, see g_variant_is_object_path
+    const QByteArray dBusCompatibleFolderId = [folder]() {
+        // replace works inplace
+        auto rv = folder->id();
+        rv.replace("-", "");
+        return rv;
+    }();
+
+    auto exporter = cloud_providers_account_exporter_new(_exporter, dBusCompatibleFolderId.constData());
```

Comment:
hm, why not 
```
const auto convertedFolderId = folder->id().replace("-", "");
..._exporter_new( _exporter, convertedFolderId.constData() );
```
?
For me that would be far more readable. And nobody would guesswork why the author choose the more elaborated approach. And could be compiled with any compiler ;-)

Reply:
Or use char's instead of strings
```
const QByteArray dBusCompatibleFolderId = folder->id().replace('-', nullptr);
````